scio-smb/src/main/scala/com/spotify/scio/smb/syntax/SortMergeBucketScioContextSyntax.scala

```diff
@@ -68,35 +70,44 @@ final class SortedBucketScioContext(@transient private val self: ScioContext) ex
     lhs: SortedBucketIO.Read[L],
     rhs: SortedBucketIO.Read[R],
     targetParallelism: TargetParallelism = TargetParallelism.auto()
-  ): SCollection[(K, (L, R))] = {
-    val t = SortedBucketIO.read(keyClass).of(lhs, rhs).withTargetParallelism(targetParallelism)
-    val (tupleTagA, tupleTagB) = (lhs.getTupleTag, rhs.getTupleTag)
-    val tfName = self.tfName
-
-    self
-      .wrap(self.pipeline.apply(s"SMB CoGroupByKey@$tfName", t))
-      .withName(tfName)
-      .applyTransform(ParDo.of(new DoFn[KV[K, CoGbkResult], (K, (L, R))] {
-        @ProcessElement
-        private[smb] def processElement(
-          @Element element: KV[K, CoGbkResult],
-          out: OutputReceiver[(K, (L, R))]
-        ): Unit = {
-          val cgbkResult = element.getValue
-          val (resA, resB) = (cgbkResult.getAll(tupleTagA), cgbkResult.getAll(tupleTagB))
-          val itB = resB.iterator()
-          val key = element.getKey
-
-          while (itB.hasNext) {
-            val b = itB.next()
-            val ai = resA.iterator()
-            while (ai.hasNext) {
-              val a = ai.next()
-              out.output((key, (a, b)))
+  ): SCollection[(K, (L, R))] = self.requireNotClosed {
+    if (self.isTest) {
+      val testInput = TestDataManager.getInput(self.testId.get)
```

Comment:
maybe we could add a static helper method to the `SMBIO` companion object to help transform `SortedBucketIO.Read`s into `testInputs` ?